### PR TITLE
Astro 971 alt

### PR DESCRIPTION
**Pull request by dmcalester**
Hey Cortney,

Providing this as an example of expectations for PRs to finish up this PI including reworking existing PRs.

In short this effort is introducing the KM color vars which should then be used to replace, wherever possible, the color properties in `_theme--dark.css` and `_theme--light.css` there are some outliers where we need to retain an old color property name. And yes, there are some inefficiencies doing it this way, but with the time remaining for this PI and supporting existing customer it’s the pragmatic approach. Once this PI is finished we can strategize about optimizing the custom props.

This draft PR made changes necessary for the dark theme button. I did not see this change made in any of the PRs from today, but may have missed one of the later efforts this afternoon, if not feel free to grab what you need.

1. Please separate PRs to one component at a time. The current merger of various tickets and components are difficult to parse and provide feedback
2. Please review your work in local Storybook prior to creating a PR. There have been several issues in the existing PRs that should have been caught prior to making a PR. If previewing on Netlify is beneficial please use the [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) feature of GitHub
3. Please use the re-request review feature after change requests have been made so reviewers know to look again at updates
4. See inline comments below for specific expectations for the code updates and edge cases. While many of the updates will be covered by a straight CSS property swap some may require additional effort to match the Astro 5/KM guidance
5. This draft PR is only for buttons, though it did remove all fallback properties in the `_theme--dark.css`

**This is not intended to be a proper commit for merging to the km branch, this is setting expectations for the KM PRs**

### CODE REVIEW

File: src/components/rux-button/README.md [hunk 1]
```diff
@@ -138,13 +138,8 @@ For more information about AstroUXDS usage outside of a Web Component environmen
...
 ## Revision History
 
-##### **TODO BEFORE RELEASE**
-
-- Test icons when they're finished, possibly change "With Icon" storybook knob to select from list of icons
-- Look at ditching class string manipulation happening in the template render function and replace with `:host`-level attributes. Keep existing class-based styles working for users of previous versions
-- Look at adding `tabindex` property or example/docs
-- Demonstrate adding click handlers/methods to buttons, possibly incorporating Storybook Actions
-- Should `size` property have a default value of `'standard'` for the button size? we don't have classes for it
+##### **5.0**
```
Comment (dmcalester) [inline src/components/rux-button/README.md:141]: All components should have a revision history update added

File: src/components/rux-button/rux-button.js [hunk 2]
```diff
@@ -122,20 +117,15 @@ export class RuxButton extends LitElement {
...
+          border-color: var(--buttonHoverBorderColor);
+          background-color: var(--buttonHoverBackgroundColor);
+          box-shadow: var(--buttonHoverBoxShadow);
         }
 
         .rux-button--outline:hover:not([disabled]) {
-          color: var(--buttonOutlineTextColor, rgb(255, 255, 255));
-          background-color: var(--buttonOutlineHoverBackgroundColor, rgba(30, 47, 66, 0.75));
-          border-color: var(--buttonOutlineHoverBorderColor, rgb(58, 129, 191));
+          color: var(--buttonOutlineHoverTextColor);
```
Comment (dmcalester) [inline src/components/rux-button/rux-button.js:126]: This is an example of where the current CSS doesn’t precisely align with the updates from KM. While the bulk of changes should be replacing a CSS property, some may require additional effort. Please test in local Storybook for adherence to colors/fonts/properties in Abstract.

File: src/components/rux-button/rux-button.js [hunk 3]
```diff
@@ -215,7 +200,7 @@ export class RuxButton extends LitElement {
       >
         <rux-icon
           icon="${this.icon}"
-          color="${this.outline ? 'rgb(0, 90, 143)' : '#ffffff'}"
+          color="${this.outline ? 'var(--primary)' : 'var(--primaryElementText)'}"
```
Comment (dmcalester) [inline src/components/rux-button/rux-button.js:203]: Another example of where more than a simple CSS swap is required.

File: src/components/rux-clock/package.json [hunk 4]
```diff
@@ -27,6 +27,6 @@
 	"publishConfig": {
 		"access": "public"
 	},
-	"version": "4.0.4",
+	"version": "4.1.0",
```
Comment (dmcalester) [inline src/components/rux-clock/package.json:30]: This change should not be part of the commit. Ignore.

File: stories/rux-button.stories.js [hunk 5]
```diff
@@ -69,7 +69,7 @@ export const slottedIconButton = () => {
         <rux-icon
           icon="custom" 
           library="/icons/custom.svg"
-          color="${outline ? 'rgb(0, 90, 143)' : '#ffffff'}"
+          color="${outline ? 'var(--primary)' : 'var(--primaryElementText)'}"
```
Comment (dmcalester) [inline stories/rux-button.stories.js:72]: Same as the component it was necessary to update the story to adapt to the KM guidance and future proof against future changes.

File: src/css/common/_theme--dark.css [hunk 6]
```diff
@@ -1,29 +1,47 @@
 .dark-theme {
+    /* 
+    
+    Astro V5 Simplified Colors
+    ==========================================================================
+
+  */
+	--backgroundColor: #101923;
+	--defaultText: #ffffff;
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:9]: I believe you addressed this in a recent commit, but `defaultText` for the first few commits was incorrect

File: src/css/common/_theme--dark.css [hunk 7]
```diff
@@ -34,39 +52,36 @@
   */
 
   /* Button */
-  --buttonTextColor: var(--colorWhite, #fff);
-  --buttonBackgroundColor: var(--colorPrimary, rgb(0, 90, 143));
-  --buttonBorderColor: var(--colorPrimary, rgb(0, 90, 143));
+  --buttonTextColor: var(--primaryElementText);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:55]: To maintain backwards compatibility and minimize the level of effort given the time remaining the intent is to use the KM var names to replace wherever possible the old var names.

File: src/css/common/_theme--dark.css [hunk 8]
```diff
@@ -34,39 +52,36 @@
...
+  --buttonHoverBackgroundColor: var(--primaryLight);
+  --buttonHoverBorderColor: var(--primaryLight);
   --buttonHoverBoxShadow: 0 8px 10px 1px rgba(0, 0, 0, 0.14),
     0 3px 14px 3px rgba(0, 0, 0, 0.12), 0 4px 5px rgba(0, 0, 0, 0.2);
 
   /* Button Active State */
-  --buttonActiveControlTextColor: var(--colorWhite, #fff);
-  --buttonActiveBackgroundColor: var(--colorPrimary, rgb(0, 90, 143));
-  --buttonActiveBorderColor: var(--colorPrimary, rgb(0, 90, 143));
+  --buttonActiveControlTextColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:69]: This is an outlier we will need to address post-PI

File: src/css/common/_theme--dark.css [hunk 9]
```diff
@@ -75,34 +90,25 @@
...
-  --controlBorderColor: var(--colorSecondaryLighten1, rgb(77, 172, 255));
-  --controlAccentColor: var(--colorSecondaryLighten1, rgb(77, 172, 255));
-
-  --controlSelectedTextColor: var(--colorWhite, #fff);
-  --controlSelectedBackgroundColor: var(
-    --colorSecondaryDarken1,
-    rgb(58, 129, 191)
-  );
-  --controlSelectedBorderColor: var(--colorWhite, rgb(255, 255, 255));
+  --controlTextColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:93]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 10]
```diff
@@ -75,34 +90,25 @@
...
     
     Progress Colors 
     ==========================================================================
 
   */
-  --progressDeterminateBarBackgroundColor: var(
-    --colorSecondary,
-    rgb(77, 172, 255)
-  );
+  --progressDeterminateBarBackgroundColor: var(--colorSecondary);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:109]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 11]
```diff
@@ -112,57 +118,47 @@
     ==========================================================================
 
   */
-  --sliderThumbBackgroundColor: var(--colorPrimary, rgb(0, 90, 143));
-  --sliderThumbBorderColor: var(--colorWhite, rgb(255, 255, 255));
+  --sliderThumbBackgroundColor: var(--primary);
+  --sliderThumbBorderColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:122]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 12]
```diff
@@ -112,57 +118,47 @@
...
   /* 
     
     Segmented Button Colors 
     ==========================================================================
 
   */
-  --segmentedButtonBorderColor: var(--colorTertiaryDarken1, rgb(30, 47, 66));
-  --segmentedButtonBackgroundColor: var(--colorPrimary, rgb(0, 90, 143));
-  --segmentedButtonTextColor: var(--colorWhite, rgb(255, 255, 255));
+  --segmentedButtonBorderColor: var(--colorTertiaryDarken1);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:137]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 13]
```diff
@@ -112,57 +118,47 @@
...
 
-  --segmentedButtonSelectedBackgroundColor: var(
-    --colorSecondaryDarken1,
-    rgb(58, 129, 191)
-  );
+  --segmentedButtonSelectedBackgroundColor: var(--colorSecondaryDarken1);
 
-  --segmentedButtonSelectedBoxShadow: inset 0 2px 4px rgba(0, 0, 0, 0.14),
-    inset 0 3px 4px rgba(0, 0, 0, 0.12), inset 0 1px 5px rgba(0, 0, 0, 0.2);
+  --segmentedButtonSelectedBoxShadow:
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:146]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 14]
```diff
@@ -172,100 +168,68 @@
...
-  --toggleBaseBorderColor: var(--colorBlack, rgb(0, 0, 0));
-  --toggleBaseTextColor: var(--colorWhite, rgb(255, 255, 255));
-
-  --toggleBaseSelectedTextColor: var(--colorNormal, rgb(86, 240, 0));
-
-  /* Alt Toggle */
-  /* --toggleThumbBorderColor: var(--colorGrayDarken2, rgb(122, 122, 122)); */
-  --toggleThumbBorderColor: #fff;
-  --toggleThumbBackgroundColor: var(--colorWhite, rgb(255, 255, 255));
+  --toggleButtonBackgroundColor: var(--primary);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:171]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 15]
```diff
@@ -172,100 +168,68 @@
...
   /* 
     
     Popup Menu Colors 
     ==========================================================================
 
   */
-  --popupMenuBackgroundColor: var(--colorWhite, rgb(255, 255, 255));
-  --popupMenuBorderColor: var(--colorSecondary, rgb(77, 172, 255));
-  --popupMenuTextColor: var(--colorBlack, rgb(0, 0, 0));
+  --popupMenuBackgroundColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:190]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 16]
```diff
@@ -172,100 +168,68 @@
...
   /* 
     
     Pushbutton Colors 
     ==========================================================================
 
   */
-  --pushbuttonBackgroundColor: var(--colorPrimary, rgb(0, 90, 143));
-  --pushbuttonBorderColor: var(--colorBlack, rgb(0, 0, 0));
-  --pushbuttonTextColor: var(--colorWhite, rgb(255, 255, 255));
+  --pushbuttonBackgroundColor: var(--primary);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:208]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 17]
```diff
@@ -172,100 +168,68 @@
...
   /* 
     
     Clock Colors 
     ==========================================================================
 
   */
-  --clockTextColor: var(--colorWhite, rgb(255, 255, 255));
-  --clockBackgroundColor: var(--colorTertiaryDarken2, rgb(20, 32, 44));
-  --clockBorderColor: var(--colorTertiary, rgb(39, 86, 128));
+  --clockTextColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:222]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 18]
```diff
@@ -172,100 +168,68 @@
...
+  --clockBorderColor: var(--colorTertiary);
 
   /* 
     
     Log Colors 
     ==========================================================================
 
   */
-  --logTextColor: var(--colorWhite, rgb(255, 255, 255));
+  --logTextColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:232]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 19]
```diff
@@ -281,35 +245,34 @@
     ==========================================================================
 
   */
-  --treeTextColor: var(--colorWhite, rgb(255, 255, 255));
-  --treeBackgroundColor: var(--colorTertiaryDarken1, rgb(32, 50, 70));
-  --treeBorderColor: var(--colorSecondaryDarken2, rgb(46, 103, 153));
-  --treeItemBorderColor: var(--colorTertiaryDarken2, rgb(24, 38, 53));
-  --treeAccentColor: var(--colorSecondary, rgb(77, 172, 255));
+  --treeTextColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:248]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 20]
```diff
@@ -281,35 +245,34 @@
...
   /* 
     
     Tab Colors 
     ==========================================================================
 
   */
-  --tabTextColor: var(--colorSecondary, rgb(77, 172, 255));
-  --tabBackgroundColor: var(--colorTertiaryDarken1, rgb(30, 47, 66));
-  --tabBorderColor: var(--colorTertiaryDarken2, rgb(20, 32, 44));
+  --tabTextColor: var(--colorSecondary);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:270]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 21]
```diff
@@ -329,16 +292,14 @@
     
   */
 
-  --inputBackgroundColor: var(--colorWhite, rgb(255, 255, 255));
+  --inputBackgroundColor: var(--colorWhite);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:295]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 22]
```diff
@@ -355,19 +316,10 @@
...
-  --scrollBarThumbBackgroundHoverColor: var(
-    --colorSecondaryDarken1,
-    rgb(58, 129, 191)
-  );
-  --scrollBarTrackBackgroundColor: var(--colorTertiaryDarken1, rgb(32, 50, 70));
-  --scrollBarTrackCornerBackgroundColor: var(
-    --colorTertiaryDarken1,
-    rgb(32, 50, 70)
-  );
+  --scrollBarThumbBackgroundColor: var(--colorSecondaryDarken2);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:319]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 23]
```diff
@@ -376,39 +328,29 @@
       
   */
 
-  --timelineHeaderBackgroundColor: var(--colorTertiaryDarken1, rgb(32, 50, 70));
-  --timelineHeaderTextColor: var(--fontColor, rgb(255, 255, 255));
+  --timelineHeaderBackgroundColor: var(--colorTertiaryDarken1);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:331]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css [hunk 24]
```diff
@@ -418,21 +360,21 @@
       
   */
 
-  --tableBorderColor: var(--colorTertiaryLighten1, rgb(83, 101, 121));
+  --tableBorderColor: var(--colorTertiaryLighten1);
```
Comment (dmcalester) [inline src/css/common/_theme--dark.css:363]: These do not reflect the KM colors. The git diff is just for the removal of the fallback color

File: src/css/common/_theme--dark.css (hunk 23 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:331]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 22 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:319]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 21 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:295]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 20 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:270]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 19 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:248]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 18 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:232]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 17 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:222]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 16 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:208]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 15 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:190]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 14 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:171]: This change is currently reflected in the Astro-971 PR - https://github.com/RocketCommunicationsInc/astro-components/pull/33

File: src/css/common/_theme--dark.css (hunk 12 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:137]: This change is currently reflected in the Astro-971 PR - #33

File: src/css/common/_theme--dark.css (hunk 13 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:146]: This change is currently reflected in the Astro-971 PR - #33

File: src/css/common/_theme--dark.css (hunk 24 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:363]: This change is currently reflected in the Astro-971 PR - #33

File: src/css/common/_theme--dark.css (hunk 6 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:9]: This change is currently reflected in the Astro-971 PR - #33

File: src/css/common/_theme--dark.css (hunk 7 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:55]: I believe this change conflicts with the design references in abstract for dark theme. Would like to discuss.

File: src/css/common/_theme--dark.css (hunk 11 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:122]: This change is currently reflected in the Astro-971 PR - #33

File: src/css/common/_theme--dark.css (hunk 10 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:109]: This change is currently reflected in the Astro-971 PR - #33

File: src/css/common/_theme--dark.css (hunk 9 above)
Comment (im-cr) [reply src/css/common/_theme--dark.css:93]: This change is currently reflected in the Astro-971 PR - #33

**Comment by im-cr** [comment]
Hey Duncan,

I've sent out a meeting sync prior to standup, I think it would help to discuss remaining ticket items. I've provided notes on the feedback above and also provided some additional notes to the bullet points at the top, thanks. 

1) Astro-971 is my primary working branch with the KM color and asset updates and global changes residing in that branch. I would ideally like to get this branch merged in asap as I am branching the smaller component updates from this branch. I'd like to discuss the component bundling but can cherry pick and push smaller PRs as needed. 

2) The storybook issue was updated in 971 and hadn't been merged into astro-922 when noted. That is also why I added the WIP label because of small discrepancies between the code, the colors list in the jira ticket Astro-971, and a few of the design mocks. All help reviewing these is appreciated and for a quicker feedback cycle I'd ask that all PRs with WIP be considered 'final' and reviewed for this block of tickets.

3) Many of the css specific changes are represented in the https://github.com/RocketCommunicationsInc/astro-components/pull/33 PR and there seem to be a few small color conflicts that I'd like to discuss on our sync call. 

I would like to discuss the last few ticket items as soon as you are available. Thanks.
